src/dom/element.rs

```diff
@@ -832,26 +841,27 @@ where
                 }))
             }
 
-            match attribute.value {
-                Some(JSXAttrValue::JSXExprContainer(JSXExprContainer {
-                    expr: JSXExpr::Expr(box ref mut expr),
-                    span,
-                })) if reserved_name_space
-                    || !matches!(expr, Expr::Lit(ref lit) if matches!(lit, Lit::Str(_) | Lit::Num(_))) =>
+            if let Some(JSXAttrValue::JSXExprContainer(JSXExprContainer {
+                expr: JSXExpr::Expr(ref mut expr),
+                span,
+            })) = &mut attribute.value
+            {
+                if reserved_name_space
+                    || !matches!(**expr, Expr::Lit(ref lit) if matches!(lit, Lit::Str(_) | Lit::Num(_)))
```

Comment:
I wonder if you could do `!matches!(**expr.is_lit(), Some(Lit::Str(_)) | Some(Lit::Num(_)))` instead